discovery-provider/src/solana/solana_program_indexer.py

```diff
@@ -228,60 +226,72 @@ def get_transaction_batches_to_process(self) -> List[List[str]]:
+                                slot={tx['slot']}, sig={tx['signature']},\
                                 latest_processed_slot(db)={latest_processed_slot}"
                             )
-                            exists = self.is_tx_in_db(read_session, tx_sig)
+                            exists = self.is_tx_in_db(read_session, tx["signature"])
                             if exists:
                                 # Exit loop and set terminal condition since this tx has been found in DB
                                 # Transactions are returned with most recently committed first, so we can assume
                                 # subsequent transactions in this batch have already been processed
                                 intersection_found = True
                                 break
                             # Otherwise, ensure this transaction is still processed
-                            transaction_signature_batch.append(tx_sig)
-                    # Restart processing at the end of this transaction signature batch
-                    last_tx = transactions_array[-1]
-                    last_tx_signature = last_tx["signature"]
-
-                    # TODO: Add Caching
-                    # Append to recently seen cache
-                    # cache_traversed_tx(redis, last_tx)
-
-                    # Append batch of processed signatures
-                    if transaction_signature_batch:
-                        transaction_signatures.append(transaction_signature_batch)
-
-                    # Ensure processing does not grow unbounded
-                    if len(transaction_signatures) > TX_SIGNATURES_MAX_BATCHES:
-                        self.msg(
-                            f"slicing tx_sigs from {len(transaction_signatures)} entries"
-                        )
-                        transaction_signatures = transaction_signatures[
-                            -TX_SIGNATURES_RESIZE_LENGTH:
-                        ]
-                        self.msg(
-                            f"sliced tx_sigs to {len(transaction_signatures)} entries"
-                        )
-
-                    # Reset batch state
-                    transaction_signature_batch = []
-
+                            unindexed_transactions.append(tx)
+                # Restart processing at the end of this transaction signature batch
+                last_tx = unindexed_transactions[-1]
+                last_tx_signature = last_tx["signature"]
+
+                # TODO: Add Caching
+                # Append to recently seen cache
+                # cache_traversed_tx(redis, last_tx)
+
+                self.msg(
+                    f"intersection_found={intersection_found},\
+                    last_tx_signature={last_tx_signature},\
+                    page_count={page_count}"
+                )
+                page_count = page_count + 1
+
+        if len(unindexed_transactions) <= FETCH_TX_SIGNATURES_BATCH_SIZE:
```

Comment:
Looks fine in terms of functionality, should we try and apply this to the existing indexers? i.e. plays/rewardmanager/user bank txs? @rickyrombo which is the most relevant one for monetization purposes

Reply:
User bank indexing